setup.py

```diff
@@ -134,7 +133,10 @@ def _determine_version(options):
     with open(version_filename, "r") as version_file:
         version_string = version_file.read().strip()
     version = packaging.version.parse(version_string)
-    if isinstance(version, packaging.version.LegacyVersion):
+    # LegacyVersion was removed in packaging 22, but is still returned by
+    # packing <= 21
+    LegacyVersion = getattr(packaging.version, "LegacyVersion", type(None))
+    if isinstance(version, LegacyVersion):
         raise ValueError("invalid version: " + version_string)
```

Comment:
This is a drive-by suggestion for something that doesn't rely on `LegacyVersion` in any way:

```
try:
   version = packaging.version.Version(version_string)
except packaging.version.InvalidVersion:
   raise ValueError("invalid version: " + version_string) from None
```

Reply:
Thanks! After the discussion in the packaging repo issue, I have been thinking about simplifying things this way, but I need to go look at the history of `Version` first and see which packaging releases it has been around for and whether the behaviour has changed over time so that I don't accidentally mess anything up for users still using older releases of packaging.

Reply:
We could even just do `version = packaging.version.Version(version_string)` because we don't care about what kind of exception is raised, just that we get an error if we try to build a package with a bad version string.

Reply:
Thanks!